### PR TITLE
add node_states to service

### PR DESCRIPTION
**Pull request by ybriffa**
this PR adds the missing node_states to the Service definition

### CODE REVIEW

**Review by ivan-savciuc — CHANGES_REQUESTED** [review]
@ybriffa, thanks for contributing. 

Would you mind adding this to an acceptance test? I see two options here: validate this part of the service using an already existing acceptance test or create a new one from scratch.

**Comment by ybriffa** [comment]
> @ybriffa, thanks for contributing.
> 
> Would you mind adding this to an acceptance test? I see two options here: validate this part of the service using an already existing acceptance test or create a new one from scratch.

Hello Ivan,
Thanks for the review, I've pushed the tests :)
Cheers

File: service.go [hunk 1]
```diff
@@ -30,6 +30,25 @@ type (
 		Integrations          []*ServiceIntegration  `json:"service_integrations"`
 		Components            []*ServiceComponents   `json:"components"`
 		Powered               bool                   `json:"powered"`
+		NodeStates            []*NodeState           `json:"node_states"`
+	}
+
+	// Node state represents the Node State model on Aiven
```
Comment (ivan-savciuc) [inline service.go:36]: Please add the `NodeState` prefix to make godoc/gofmt happy.

File: service.go [hunk 2]
```diff
@@ -30,6 +30,25 @@ type (
...
+	}
+
+	// ProgressUpdate state represents the Progress Update model on Aiven
+	ProgressUpdate struct {
+		Completed bool   `json:"completed"`
+		Current   int    `json:"current"`
+		Max       int    `json:"max"`
+		Min       int    `json:"min"`
+		Phase     string `json:"phase"`
+		Unit      string `json:"phase"`
```
Comment (ivan-savciuc) [inline service.go:51]: Perhaps a typo here: Unit has `json:"phase"` JSON notation

**Review by ivan-savciuc — CHANGES_REQUESTED** [review]
@ybriffa, thanks for adding tests. I found these small things otherwise, LGTM.

File: service.go (hunk 2 above)
Comment (ybriffa) [reply service.go]: indeed, I've just fixed it :)

**Review by ivan-savciuc — APPROVED** [review]

**Pull request merged**